File: 1.computer-science/1.DS-and-Algorithms/5.competitive-programming/#leetcode/657. Robot Return to Origin.ts
//https://leetcode.com/problems/robot-return-to-origin
function judgeCircle(moves: string): boolean {
    let x = 0, y = 0;
    for (const move of moves) {
        if (move === 'R') x++;
        else if (move === 'L') x--;
        else if (move === 'U') y++;
        else if (move === 'D') y--;
    }
    return !x && !y;
}

console.log(judgeCircle("UD")) //true
console.log(judgeCircle("LL")) //false
console.log(judgeCircle("RRDD")) //false

File: 1.computer-science/1.DS-and-Algorithms/5.competitive-programming/#leetcode/657. Robot Return to Origin.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool judgeCircle(string moves) {
        int x = 0, y = 0;
        for (char move : moves) {
            if (move == 'R') x++;
            else if (move == 'L') x--;
            else if (move == 'U') y++;
            else if (move == 'D') y--;
        }
        return x == 0 && y == 0;
    }
};



int main() {
    Solution sol;
    string input = "UDLR";
    bool result = sol.judgeCircle(input);
    cout << (result ? "true" : "false") << endl;
    return 0;
}

